tests/src/bignum_helpers.c

```diff
@@ -85,6 +85,23 @@ int mbedtls_test_read_mpi_core( mbedtls_mpi_uint **pX, size_t *plimbs,
     return( MBEDTLS_ERR_MPI_BAD_INPUT_DATA );
 }
 
+int mbedtls_test_read_mpi_modulus( mbedtls_mpi_mod_modulus *N,
+                                   const char *s,
+                                   mbedtls_mpi_mod_rep_selector int_rep )
+{
+    mbedtls_mpi_uint *p = NULL;
+    size_t limbs = 0;
+    if( N->limbs != 0 )
+        return( MBEDTLS_ERR_MPI_BAD_INPUT_DATA );
+    int ret = mbedtls_test_read_mpi_core( &p, &limbs, s );
+    if( ret != 0 )
+        return( ret );
+    ret = mbedtls_mpi_mod_modulus_setup( N, p, limbs, int_rep );
```

Comment:
While hiding this logic in a helper function is helpfull, I cannot help but wonder weather we need a simple, `mbedtls_mpi_mod_modulus_is_setup()` method to test and fail. 

This could ensure no operation takes place if the passed in moduli has been setup. 

This could fall outside of the scope of this PR and could be looked as part of the #6757

Reply:
Apologies for not being clear.

In the current form it is possible to do the following


```
mbedtls_mpi_uint *p1;
mbedtls_mpi_uint *p2;
........
........
mbedtls_mpi_mod_modulus_init(&N);
ret = mbedtls_mpi_mod_modulus_setup( N, p1, limbs, int_rep );
........
ret = mbedtls_mpi_mod_modulus_setup( N, p2, limbs, int_rep );
```

So in a way the setup method can be used to update a modulus with a different set of data. That is not acceptable in the design. 

So I wanted to capture that we could have a simple helper method

```
int mbedtls_mpi_mod_modulus_is_setup( mbedtls_mpi_mod_modulus *m )
{
    if( m.p == NULL && m->limbs + m->bits == 0 && m->int_rep == MBEDTLS_MPI_MOD_REP_INVALID )
        return 0
    return 1
}
```

To check. 

But that is outside of the scope of this PR so you may resolve this.

Reply:
Ah, I see. Arguably `mbedtls_mpi_mod_modulus_setup` should check whether the modulus is already set up and reject resetting, because that's an unusual thing to do which can invalidate existing residues, and callers that are sure they can do that should free the modulus first. I'm not convinced that `mbedtls_mpi_mod_modulus_is_setup` is a particularly useful building block: where else would we call it? Anyway, I would consider improving the test helper functions as in scope here, but not improving `mbedtls_mpi_mod_modulus_setup` itself. Filed as https://github.com/Mbed-TLS/mbedtls/issues/6800